### DISCUSSION TITLE
Coupling fabm with fabm-deb model?

### DISCUSSION THREAD

**Discussion by tongyaopu** [fabm-model/fabm#38] [Q&A] [answered]
Hi,

I am trying to couple FABM 0d driver with the FABM-DEB model. Should I specify the driver (0d) that the DEB model should be compiled to? This somehow wiped out the links to all default biogeochemistry models (e.g. npzd. `npzd: "gotm/npzd" is not a valid model name.`)

```

cd ~/build/fabm

FABMDIR="~/src/fabm" 	# Path to FABM source code
GOTMDIR="~/src/GOTM6" 	# Path to GOTM source code 
compiler="/opt/homebrew/bin/gfortran" 	# fortran compiler on this computer


# >>>> compile with FABM? >>>
cmake $FABMDIR -DCMAKE_FORTRAN_COMPILER=$compiler -DFABM_HOST=GOTM
cmake $FABMDIR -DCMAKE_FORTRAN_COMPILER=$compiler -DFABM_HOST=0d -DGOTM_BASE=$GOTMDIR 
cmake $FABMDIR -DCMAKE_FORTRAN_COMPILER=$compiler -DFABM_HOST=python

# compile deb model?
cmake $FABMDIR  -DCMAKE_FORTRAN_COMPILER=$compiler -DFABM_HOST=0d -DGOTM_BASE=$GOTMDIR  -DFABM_INSTITUTES=deb -DFABM_DEB_BASE=/Users/tongyaop/src/fabm-deb

make install
```

**Comment by jornbr** [comment, marked as answer]
To build the 0D driver you only need a single cmake command. You can make it include additional biogeochemical models such as fabm-deb with command line options `-DFABM_INSTITUTES` and `-DFABM_DEB_BASE` as you have been doing. In your case, this may work:

```
cd ~/build/fabm0d

FABMDIR="~/src/fabm" 	# Path to FABM source code
GOTMDIR="~/src/GOTM6" 	# Path to GOTM source code 
compiler="/opt/homebrew/bin/gfortran" 	# fortran compiler on this computer

cmake $FABMDIR/src/drivers/0d -DGOTM_BASE=$GOTMDIR -DCMAKE_Fortran_COMPILER=$compiler -DFABM_INSTITUTES=deb -DFABM_DEB_BASE=/Users/tongyaop/src/fabm-deb
make install
```

However, `FABM_INSTITUTES` specifies the full set of biogeochemical models that need to be included. The above `-DFABM_INSTITUTES=deb` _only_ includes the DEB model; as you have seen, all other biogeochemical models are excluded. By default `FABM_INSTITUTES` is a long list: `akvaplan;au;bb;csiro;ersem;examples;gotm;iow;jrc;msi;niva;pclake;pml;selma;su;uhh`. If you want all those defaults _plus_ the DEB model, you'd specify: `-DFABM_INSTITUTES="akvaplan;au;bb;csiro;ersem;examples;gotm;iow;jrc;msi;niva;pclake;pml;selma;su;uhh;deb"` (note the `deb` at the end)

**Reply by tongyaopu** [reply]
Thank you for the reply! This has worked for me
